src/DistortableCollection.js

```diff
@@ -199,19 +199,65 @@ L.DistortableCollection = L.FeatureGroup.extend({
     return reduce / imgs.length;
   },
 
-  isJsonDetected(currentURL) {
+  _extractJSONDownloadURL(currentURL) {
+    const startIndex = currentURL.lastIndexOf('=');
+    const jsonDownloadURL = currentURL.slice(startIndex + 1);
+
+    return jsonDownloadURL;
+  },
+
+  _isJsonDetected(currentURL) {
     if (currentURL.includes('?json=')) {
-      startIndex = currentURL.lastIndexOf('.');
-      fileExtension = currentURL.slice(startIndex + 1);
+      const startIndex = currentURL.lastIndexOf('.');
+      const fileExtension = currentURL.slice(startIndex + 1);
 
       if (fileExtension === 'json') {
-        console.log('JSON found in map shareable link');
+        console.log('JSON found in map shareable link'); // left here purposely
         return true;
       }
     }
+
     return false;
   },
 
+  async _generateFromJson(jsonDownloadURL) {
+    let index = 0;
+    const imgCollectionProps = [];
+
+    try {
+      const response = await axios.get(jsonDownloadURL);
+      if (response.data.images.length > 1) {
+        response.data.images.forEach((data) => {
+          imgCollectionProps[index] = data;
+          index++;
+        });
+        return {avg_cm_per_pixel: response.data.avg_cm_per_pixel, imgCollectionProps};
+      }
+      imgCollectionProps[index] = response.data.images;
+
+      return {avg_cm_per_pixel: response.data.avg_cm_per_pixel, imgCollectionProps};
+    } catch (err) {
+      console.log('err', err);
+    }
+  },
+
+  // expects URL in this format: 'http://localhost:8081/examples/shareable.html?json=https://archive.org/download/segeotest/segeotest.json'
+  async generateFromJsonUrl(currentURL) {
```

Comment:
Here, I think we should be doing this in the examples folder, since another application using this library might already be using the URL for its purposes.

Reply:
Okay, thanks.